runtime/src/chpl-init.c

```diff
@@ -334,8 +338,11 @@ void chpl_execute_module_deinit(c_fn_ptr deinitFun) {
 // }
 //
 void chpl_library_init(int argc, char* argv[]) {
-    chpl_rt_init(argc, argv);                   // Initialize the runtime
+  chpl_rt_init(argc, argv);                     // Initialize the runtime
   chpl_task_callMain(chpl_std_module_init);     // Initialize std modules
+  chpl_libraryModuleLevelSetup();
+
+  // TODO: Call chpl_rt_preUserCodeHook() here for Locale[0]?
```

Comment:
Should this TODO still be here?

Reply:
I'm not sure, I think it might be good to leave it for now, at least in case we run into problems with comm barriers on multilocale libraries or something...

Reply:
I'm not certain if we do.  I'm okay with leaving it in, but it would be good to add a date and your name to the TODO so folks can know who to ask about it later